starknet_devnet/starknet_wrapper.py

```diff
@@ -450,7 +481,11 @@ async def invoke(self, external_tx: InvokeFunction):
     async def call(self, transaction: CallFunction):
         """Perform call according to specifications in `transaction`."""
 
-        state_copy = self.get_state().copy()
+        if self.config.blocks_on_demand and self.mempool_state is not None:
+            state_copy = self.mempool_state
```

Comment:
Shouldn't this be `self.mempool_state.copy()`? E.g. if a call was made twice in a row, wouldn't it produce different results since `mempool_state` was actually altered?

Reply:
but mempool state is already a copy of state:
```
        # Update mempool state before block generation
        self.__mempool_state = self.get_state().copy()
```
so now it will be referenced to variable `state_copy` and a call will be made.